src/wasm-type.h

```diff
@@ -26,21 +26,29 @@
 // Signature, Struct and Array types that will be single but not basic. To
 // prepare for this change, the following macro marks affected code locations.
 #define TODO_SINGLE_COMPOUND(type)                                             \
-  assert(!type.isMulti() && "Unexpected multi-value type");                    \
+  assert(!type.isTuple() && "Unexpected tuple type");                          \
   assert(!type.isCompound() && "TODO: handle compound types");
 
 namespace wasm {
 
+class Type;
+struct TypeDef;
```

Comment:
Since `TypeDef` was moved to the .cpp file, I don't think this should be necessary. The overloads that still use `TypeDef` below in this file can also be moved to the .cpp file.